### ISSUE TITLE
XCFramework failing to build for macOS but logs have "ARCHIVE SUCCEEDED"

### ISSUE THREAD

**Issue by BenShutt** [Carthage/Carthage#3251]
* carthage install method: [ ] .pkg, [x] homebrew, [ ] source
* `which carthage`: `/usr/local/bin/carthage`
* `carthage version`: 0.38.0
* `xcodebuild -version`: Xcode 13.2.1
* Are you using `--no-build`? No
* Are you using `--no-use-binaries`? No (building from source anyway)
* Are you using `--use-submodules`? No
* Are you using `--cache-builds`? No
* Are you using `--new-resolver`? No
* Are you using `--use-xcframeworks`? Yes

**Cartfile**
```
github "3sidedcube/HTTPRequest" "release/v1.0.0"
```

**Command**
```
carthage update --use-xcframeworks --platform macos
```

**Carthage Output**
```
*** Fetching HTTPRequest
*** Fetching Alamofire
*** Checking out Alamofire at "5.5.0"
*** Checking out HTTPRequest at "a8c760713cca0e2f4cc760daf3ea0de4daab9d78"
*** xcodebuild output can be found in /var/folders/4w/tq9d3_h54_x6wg_jt4crxxxr0000gn/T/carthage-xcodebuild.MvUjz8.log
*** Building scheme "Alamofire macOS" in Alamofire.xcworkspace
*** Building scheme "HTTPRequest" in HTTPRequest.xcodeproj
Build Failed
	Task failed with exit code 1:
	/usr/bin/xcrun dsymutil /Users/user/Library/Caches/org.carthage.CarthageKit/DerivedData/13.2.1_13C100/HTTPRequest/a8c760713cca0e2f4cc760daf3ea0de4daab9d78/Build/Intermediates.noindex/ArchiveIntermediates/HTTPRequest/BuildProductsPath/Release-iphoneos/HTTPRequest.framework/HTTPRequest -o /Users/user/Library/Caches/org.carthage.CarthageKit/DerivedData/13.2.1_13C100/HTTPRequest/a8c760713cca0e2f4cc760daf3ea0de4daab9d78/Build/Intermediates.noindex/ArchiveIntermediates/HTTPRequest/BuildProductsPath/Release-iphoneos/HTTPRequest.framework.dSYM

This usually indicates that project itself failed to compile. Please check the xcodebuild log for more details: /var/folders/4w/tq9d3_h54_x6wg_jt4crxxxr0000gn/T/carthage-xcodebuild.MvUjz8.log
```

**Actual outcome**
Carthage fails to build the `xcframework`. However, going into the log at the end it has `*** ARCHIVE SUCCEEDED ***`.
It works using SPM. Also there is a command line program in the project referencing this which runs on macOS and works fine.
It builds when using `--platform ios`.

**Expected outcome**
Carthage should build the `xcframework`

**Comment by tmspzz** [comment]
have you checked the build log for what is failing? 

Seems like `dsymutil` is encountering an error when dumping symbol to file. You can executer the same command manually to find out what's wrong.

**Comment by BenShutt** [comment]
Thanks for getting back to me @tmspzz. 😄 

Yep, I have looked at the log file. I couldn't see any logs obviously spitting out errors or issues.
It is pretty massive though so I may have missed something.

These two could potentially be causing pain, but unlikely:
```
--- xcodebuild: WARNING: Using the first of multiple matching destinations:
{ platform:macOS, arch:x86_64, id:<mac-uuid> }
{ platform:macOS, name:Any Mac }
```
or
```
remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
```

I saved the log file after raising this issue so happy to share it provided there's nothing sensitive in there (my name ofc is fine). It's still odd that it's saying `ARCHIVE SUCCEEDED`.

I suspect it's down to the project build settings in some way if we are not seeing others with this issue. Pretty confident the Swift is compiling fine.

Running the following (after running the `carthage update --use-xcframeworks --platform macos`):
```
dsymutil /Users/.../HTTPRequest -o /Users/.../HTTPRequest.framework.dSYM
```
doesn't report an error, the terminal command simply returns. (The Alamofire command also behaves this way).
